jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaDebugSupport.kt

```diff
@@ -0,0 +1,125 @@
+
+class JavaRuntimeDebugSupport : RuntimeDebugSupport {
+    override suspend fun createDebugProcess(
+        context: Context,
+        environment: ExecutionEnvironment,
+        state: SamRunningState,
+        debugHost: String,
+        debugPorts: List<Int>
+    ): XDebugProcessStarter = createDebugProcess(environment, debugHost, debugPorts)
+}
+
+abstract class JavaImageDebugSupport : ImageDebugSupport {
+    override fun supportsPathMappings(): Boolean = true
+    override val languageId = JavaLanguage.INSTANCE.id
+    override suspend fun createDebugProcess(
+        context: Context,
+        environment: ExecutionEnvironment,
+        state: SamRunningState,
+        debugHost: String,
+        debugPorts: List<Int>
+    ): XDebugProcessStarter = createDebugProcess(environment, debugHost, debugPorts)
+}
+
+open class Java8ImageDebugSupport : JavaImageDebugSupport() {
+    override val id: String = LambdaRuntime.JAVA8.toString()
+    override fun displayName() = LambdaRuntime.JAVA8.toString().capitalize()
+    override fun containerEnvVars(debugPorts: List<Int>): Map<String, String> = mapOf(
+        "_JAVA_OPTIONS" to "-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=${debugPorts.first()} " +
+            "-XX:MaxHeapSize=2834432k -XX:MaxMetaspaceSize=163840k -XX:ReservedCodeCacheSize=81920k -XX:+UseSerialGC " +
+            "-XX:-TieredCompilation -Djava.net.preferIPv4Stack=true -Xshare:off"
+    )
+}
+
+class Java8Al2ImageDebugSupport : Java8ImageDebugSupport() {
+    override val id: String = LambdaRuntime.JAVA8_AL2.toString()
+    override fun displayName() = LambdaRuntime.JAVA8_AL2.toString().capitalize()
+}
+
+open class Java11ImageDebugSupport : JavaImageDebugSupport() {
+    override val id: String = LambdaRuntime.JAVA11.toString()
+    override fun displayName() = LambdaRuntime.JAVA11.toString().capitalize()
+    override fun containerEnvVars(debugPorts: List<Int>): Map<String, String> = mapOf(
+        "_JAVA_OPTIONS" to "-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=*:${debugPorts.first()} " +
+            "-XX:MaxHeapSize=2834432k -XX:MaxMetaspaceSize=163840k -XX:ReservedCodeCacheSize=81920k -XX:+UseSerialGC " +
+            "-XX:-TieredCompilation -Djava.net.preferIPv4Stack=true"
+    )
+}
+
+private val edtContext = getCoroutineUiContext()
+
+private suspend fun createDebugProcess(
+    environment: ExecutionEnvironment,
+    debugHost: String,
+    debugPorts: List<Int>
+): XDebugProcessStarter {
+    val connection = RemoteConnection(true, debugHost, debugPorts.first().toString(), false)
+    val debugEnvironment = RemotePortDebugEnvironment(environment, connection)
+    val debuggerManager = DebuggerManagerEx.getInstanceEx(environment.project)
+
+    val debuggerSession = withContext(edtContext) {
+        debuggerManager.attachVirtualMachine(debugEnvironment)
+    } ?: throw IllegalStateException("Attaching to the JVM failed")
```

Comment:
can we add the connection to the exception (does it have a toString?)

Reply:
Connection does not have a tostring, but added printing the host and port